WordPress/src/main/java/org/wordpress/android/ui/posts/PostsListFragment.java

```diff
@@ -661,17 +743,55 @@ public void onPostUploaded(OnPostUploaded event) {
         }
     }
 
+    @SuppressWarnings("unused")
+    @Subscribe(threadMode = ThreadMode.MAIN)
+    public void onPostsSearched(OnPostsSearched event) {
+        mIsSearching = false;
+        hideLoadMoreProgress();
+
+        if (event.isError()) {
+            AppLog.w(AppLog.T.POSTS, "Error searching posts: " + event.error.message);
+            ToastUtils.showToast(getActivity(), "Error searching posts: " + event.error.type);
+        } else {
+            if (event.searchResults != null && !ListUtils.isEmpty(event.searchResults.getPosts())) {
+                hideEmptyView();
+                mSearchResults.addAll(event.searchResults.getPosts());
+                setLocalPostIdsOnSearchResults();
+                getPostListAdapter().setPostList(mSearchResults);
```

Comment:
This is also true for the `else` statement below and everything we do in the callbacks. We should also check the regular fetch post callbacks too, since they can also override changes from search.